UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -404,86 +404,90 @@ protected IEnumerator ConstructBuffer(GLTF.Schema.Buffer buffer, int bufferIndex
             }
         }
 
-        protected IEnumerator ConstructImage(GLTF.Schema.Image image, int imageCacheIndex, bool markGpuOnly = true)
+        protected byte[] GetImageDataBuffer(GLTF.Schema.Image image, int imageCacheIndex)
         {
-            if (_assetCache.ImageCache[imageCacheIndex] == null)
+            if (image.BufferView != null)
+            {
+                // Read from GLB
+                var bufferView = image.BufferView.Value;
+                var data = new byte[bufferView.ByteLength];
+                var bufferContents = _assetCache.BufferCache[bufferView.Buffer.Id];
+                bufferContents.Stream.Position = bufferView.ByteOffset + bufferContents.ChunkOffset;
+                bufferContents.Stream.Read(data, 0, data.Length);
+                return data;
+            }
+            else
             {
-                if (image.BufferView != null)
+                string uri = image.Uri;
+                byte[] bufferData;
+                URIHelper.TryParseBase64(uri, out bufferData);
+                if (bufferData != null)
                 {
-                    yield return ConstructImageFromGLB(image, imageCacheIndex);
+                    return bufferData;
                 }
                 else
                 {
-                    string uri = image.Uri;
-
-                    byte[] bufferData;
-                    URIHelper.TryParseBase64(uri, out bufferData);
-                    if (bufferData != null)
+                    Stream stream = _assetCache.ImageStreamCache[imageCacheIndex];
+                    if (stream is MemoryStream)
                     {
-                        Texture2D loadedTexture = new Texture2D(0, 0);
-                        loadedTexture.LoadImage(bufferData, true);
+                        using (MemoryStream memoryStream = stream as MemoryStream)
+                        {
 
-                        _assetCache.ImageCache[imageCacheIndex] = loadedTexture;
-                        yield return null;
+                            return memoryStream.ToArray();
+                        }
                     }
                     else
                     {
-                        Stream stream = _assetCache.ImageStreamCache[imageCacheIndex];
-                        yield return ConstructUnityTexture(stream, markGpuOnly, image, imageCacheIndex);
+                        byte[] buffer = new byte[stream.Length];

```

Comment:
the data originally comes from a stream, so it seems inefficient to pass it back into one